extern/sector-storage/fsutil/filesize_unix.go

```diff
@@ -11,19 +11,24 @@ type SizeInfo struct {
 	OnDisk int64
 }
 
-// FileSize returns bytes used by a file on disk
+// FileSize returns bytes used by a file or directory on disk
 func FileSize(path string) (SizeInfo, error) {
-	var stat syscall.Stat_t
-	if err := syscall.Stat(path, &stat); err != nil {
-		if err == syscall.ENOENT {
+	var size int64
+	err := filepath.Walk(path, func(_ string, info os.FileInfo, err error) error {
+		if err != nil {
+			return err
+		}
+		if !info.IsDir() {
+			size += info.Size()
```

Comment:
.Size() doesn't return allocated bytes, which is what this method returned previously (Can be done by using `info.Sys().(syscall.Stat_t)...`). In case some of those files are preallocated, we should be returning allocated bytes, otherwise we may report too little reserved space relative to what the filesystem reports as free